frontend/src/components/organisms/ManageUserProfile.tsx

```diff
@@ -13,21 +13,128 @@ import LinearProgress from "@mui/material/LinearProgress";
 import ArrowBackIosNewIcon from "@mui/icons-material/ArrowBackIosNew";
 import { IconButton } from "@mui/material";
+import { BASE_URL } from "@/utils/constants";
+import { auth } from "@/utils/firebase";
 
-interface ManageUserProfileProps {
+type userProfileData = {
+  name: string;
+  role: string;
+  email: string;
+  joinDate: string;
   userid: string;
+  hours: number;
+  status: string;
+};
+
+interface ManageUserProfileProps {
+  userProfileDetails: userProfileData;
+}
+
+type userStatusData = {
+  userRole: string;
+  userStatus: string;
+  userID: string;
+};
+
+type userRegistrationData = {
+  totalHours: number;
+  userRegistrations: any[];
+};
+
+type verifyData = {
+  totalHours: number;
+};
+
+function formatDateString(dateString: string) {
+  const date = new Date(dateString);
+  const month = (date.getMonth() + 1).toString().padStart(2, "0");
+  const day = date.getDate().toString().padStart(2, "0");
+  const year = date.getFullYear();
+  return `${month}/${day}/${year}`;
 }
 
 /**
  * A ManageUserProfile component
  */
-const Status = () => {
-  const [role, setRole] = React.useState("Volunteer");
+const Status = ({ userRole, userStatus, userID }: userStatusData) => {
+  const [role, setRole] = React.useState(userRole);
+  const [status, setStatus] = React.useState(userStatus);
 
-  const handleChange = (event: SelectChangeEvent) => {
+  const handleChange = async (event: SelectChangeEvent) => {
+    //call the PATCH request here to change the ROLE
+    try {
+      const url = BASE_URL as string;
+      const fetchUrl = `${url}/users/` + userID + `/role`;
+      const currentUser = auth.currentUser;
+      const userToken = await currentUser?.getIdToken();
+
+      const val = event.target.value;
+      // CHANGES THE USER ROLE
+      const body = { role: val };
+      const response = await fetch(fetchUrl, {
+        method: "PATCH",
+        headers: {
+          Authorization: `Bearer ${userToken}`,
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify(body),
+      });
+      if (response.ok) {
+        console.error("User Role Changed Successfully", response.status);
+        const data = await response.json();
+      } else {
+        console.error("User PATCH failed with status:", response.status);
+      }
+    } catch (error) {
+      console.log("Error in PATCH.");
+      console.log(error);
+    }
     setRole(event.target.value);
   };
 
+  const handleClick = async () => {
+    // this stores what value the Status of the user is
+    var bodyval = "";
+    try {
+      const url = BASE_URL as string;
+      const fetchUrl = `${url}/users/` + userID + `/status`;
+      const currentUser = auth.currentUser;
+      const userToken = await currentUser?.getIdToken();
+
+      // BLACKLISTS THE USER
+      if (status == "ACTIVE") {
+        bodyval = "HOLD";
+      } else if (status == "HOLD") {
+        bodyval = "ACTIVE";
+      } else {
+        bodyval = "INACTIVE";
+      }
```

Comment:
I think once you address the incompleteness of the blacklist user flow this will become unnecessary!